### PR TITLE
Enable auto-generation of CONFIG.yaml

### PR DESCRIPTION
**Pull request by parth-io**
Also added new parameters --no-save and --reset-configuration
Updated README.md

### CODE REVIEW

File: src/login.js [hunk 1]
```diff
@@ -90,8 +87,8 @@ async function login() {
 
     const access_token = res.data.access_token;
     if (access_token === undefined) throw 'Login failed, please try again. LumiNUS sucks sometimes.';
-    if (PRINT) console.log('done!');
+    if (readPrint()) console.log('done!');
     return 'Bearer ' + access_token;
 }
 
-module.exports = { login };
+module.exports = { login, getUsername };
```
Comment (halfwhole) [inline src/login.js]: Any reason for exporting `getUsername` as well?

**Review by halfwhole — COMMENTED** [review]
This looks good, though the error message when the user's login credentials are invalid will probably have to be changed (I can do this myself!) -- it's to remind the user that they have to use `-r` to reset them

File: src/login.js (hunk 1 above)
Comment (parth-io) [reply src/login.js]: That can be safely deleted. It was for running the tests

**Review by halfwhole — APPROVED** [review]
LGTM, thanks for contributing! Sorry for taking a while.
I'll make the aforementioned minor changes to the error messages soon, and update the README as well.

**Pull request merged**